ScienceDirect.js

```diff
@@ -69,23 +69,45 @@ function getPDFLink(doc, onDone) {
 	// If intermediate page URL is available, use that directly
 	var intermediateURL = attr(doc, '.PdfEmbed > object', 'data');
 	if (intermediateURL) {
-		//Zotero.debug("Embedded intermediate URL: " + intermediateURL);
+		//Zotero.debug("Embedded intermediate PDF URL: " + intermediateURL);
 		parseIntermediatePDFPage(intermediateURL, onDone);
 		return;
 	}
 	
-	// For pages that use the new JS drop-down, the DOM doesn't seem to have the intermediate page URL
-	// after the page load, but it's in the initial HTML, so fetch the page again. This is awful, and
-	// we should try to find a better way of getting the URL (which is obviously available to JS in
-	// the page).
+	// Simulate a click on the "Download PDF" button to open the menu containing the link with the URL
+	// for the intermediate page, which doesn't seem to be available in the DOM after the page load.
+	// This is an awful hack, and we should look out for a better way to get the URL, but it beats
+	// refetching the original source as we do below.
+	var pdfLink = doc.querySelector('#pdfLink');
+	if (pdfLink) {
+		// Just in case
+		try {
+			pdfLink.click();
+			intermediateURL = attr(doc, '.PdfDropDownMenu li a', 'href');
+			var clickEvent = document.createEvent('MouseEvents');
```

Comment:
Now, it works fine! (FYI I am testing with Scaffold and Zotero 4.0 in an old Firefox.)